### PR TITLE
Translation keys 2

### PR DESCRIPTION
**Pull request by benalleng**
Another pass at translations, this one has a large focus on settings

additionally a small error was fixed in `amountEditable.tsx` so that only holding the DEL key will erase the entire amount

### CODE REVIEW

**Comment by futurepaul** [comment]
I think you need a rebase? seeing dupe changes

**Review by futurepaul — APPROVED** [review]
sweet

File: src/components/DeleteEverything.tsx [hunk 1]
```diff
@@ -34,7 +36,14 @@ export function DeleteEverything(props: { emergency?: boolean }) {
                 await MutinyWallet.import_json("{}");
             }
 
-            showToast({ title: "Deleted", description: `Deleted all data` });
+            showToast({
+                title: `${i18n.t(
```
[the remark this answers is not in the document]
Comment (benalleng) [reply src/components/DeleteEverything.tsx:40]: Oops that's a mistake! that doesn't need to be string interpolation
```
<SettingsCard
                    title={`${i18n.t(
                        "settings.connections.manage_connections"
                    )}`}
                >
```

This however does need to be string interpolation

[the remark this answers is not in the document]
Comment (benalleng) [reply src/components/DeleteEverything.tsx]: also for some reason the captions for `SettingsLinkList` in `/settings/index.ts` need to be as well but the text doesn't, I think it has something to do with the fact that captions are an optional param

[the remark this answers is not in the document]
Comment (benalleng) [reply src/components/DeleteEverything.tsx]: fixed for that caption, I allowed it to be a string or the i18next class `DefaultTFuncReturn`. not sure if this is something we would want to implement across the codebase or just stick with string interpolation when needed

File: src/routes/settings/index.tsx [hunk 2]
```diff
@@ -10,13 +10,15 @@ import NavBar from "~/components/NavBar";
...
+import { useI18n } from "~/i18n/context";
+import { DefaultTFuncReturn } from "i18next";
 
 function SettingsLinkList(props: {
     header: string;
     links: {
         href: string;
         text: string;
-        caption?: string;
+        caption?: string | DefaultTFuncReturn;
```
Comment (futurepaul) [inline src/routes/settings/index.tsx:21]: looks like this type stuff can be solved with

https://www.i18next.com/overview/typescript#argument-of-type-defaulttfuncreturn-is-not-assignable-to-parameter-of-type-xyz

https://stackoverflow.com/questions/74143213/how-to-handle-i18next-new-type-defaulttfuncreturn

**Review by futurepaul — CHANGES_REQUESTED** [review]
seems like there's a good solve for the ts stuff

**Comment by futurepaul** [comment]
played around a bit with typescript stuff mentioned in here: https://www.i18next.com/overview/typescript. ideally we can make it a type / lint error if there's a translation reference that's missing a default translation... but I haven't had luck yet (it shows them all as missing, like in the screenshot). don't need to block on it but just something to be aware of that we can move towards

Screenshot 2023-07-26 at 5 25 09 PM

here's my i18next.d.ts file:

```
import { resources, defaultNS } from "~/i18n/config";

declare module "i18next" {
    interface CustomTypeOptions {
        defaultNS: typeof defaultNS;
        resources: (typeof resources)["en"];
        returnNull: false;
    }
}
```

and I edited the config like this:

```
export const resources = {
    en: {
        translations: en
    },
    pt: {
        translations: pt
    }
};

export const defaultNS = "translations";
```

**Comment by benalleng** [comment]
So I am getting in-line errors when there is an incorrect key <=> namespace match but I am also not getting any sort of lint errors. I ended up catching a few incorrect keys as a result.

In-line error
`No overload matches this call.  Overload 1 of 12, '(key: TemplateStringsArray | "translations:common" | ...)'`

File: src/i18n/en/translations.ts [hunk 3]
```diff
@@ -34,23 +76,160 @@ export default {
...
+            emergency_link: "emergency kit."
+        },
+        not_found: {
+            title: "Not Found",
+            wtf_paul: "This is probably Paul's fault."
+        }
+    },
     create_an_issue: "Create an issue",
-    feedback: "Bugs? Feedback?",
+    FFfeedback: "Bugs? Feedback?",
```
Comment (futurepaul) [inline src/i18n/en/translations.ts:222]: typo? also where is this used?

Comment (benalleng) [reply src/i18n/en/translations.ts:222]: actually no, I added those to get rid of an error caused by the repeat `feedback` key and set myself to look for it later but never did :sweat_smile:. I don't think it is used, I think it is an artifact from before #378 and i never removed it

**Review by futurepaul — APPROVED** [review]
heck yeah... let's try and figure out this linter thing soon I think that will be a big help for catching misses

**Pull request merged**